src/main/java/com/google/impactdashboard/database_manager/data_read/DataReadManagerImpl.java

```diff
@@ -124,6 +197,30 @@ public double getAverageIAMBindingsInPastYear(String projectId) {
     return datesToBindings;
   }
 
+  /** 
+   *  Returns a map of dates (as timestamps in UTC milliseconds since the epoch) 
+   *  to the number of IAM bindings that existed on that date, summed across all projects
+   *  belonging to the organization with id {@code organizationId}. 
+   */
+  @Override
+  public Map<Long, Integer> getOrganizationDatesToBindings(String organizationId) {
```

Comment:
Fyi, when adding methods that contains large amount of duplicated code, can consider extract common parts out into a separate methods so that it's easy to maintain in the long term.

Reply:
I went in and factored out as much as I could!